src/communication/bluetooth/chunker/BluetoothChunker.cpp

```diff
@@ -0,0 +1,164 @@
+#include "communication/bluetooth/chunker/BluetoothChunker.h"
+
+BluetoothChunker::BluetoothChunker(BLEService* pService, const char* cuuid, const uint32_t access) {
+  characteristic = pService->createCharacteristic(cuuid, access);
+  BLECallback = new BLEChunkerCallback(this);
+  characteristic->setCallbacks(BLECallback);
+};
+
+BluetoothChunker::~BluetoothChunker() {
+  delete characteristic;
+  delete BLECallback;
+}
+
+bool BluetoothChunker::startReceiving(std::string& headerChunk) {
+  if(isActiveReceiving()) {
+    Logger::error("[BluetoothChunker::startReceiving()] Reveiving transaction already started");
+    return false;
+  }
+  
+  if(headerChunk.length() < headerSize) {
+    Logger::error("[BluetoothChunker::startReceiving()] Invalid message");
+    return false;
+  }
+
+  toReceiveChunkCount = 0;
+  //copy num save in many chars to single long type variable
+  for(int i = 0; i < headerSize; i++) {
+    // save sizeof(char) bytes
+    toReceiveChunkCount = toReceiveChunkCount | headerChunk[i]; 
```

Comment:
haha I would love to see here something like `Long.fromBytes(headerChunk[:headerSize])` but I guess cpp doesn't have that feature